vyper/context/types/function.py

```diff
@@ -175,16 +211,20 @@ def from_FunctionDef(
                 kwargs["nonreentrant"] = decorator.args[0].value
 
             elif isinstance(decorator, vy_ast.Name):
-                if decorator.id in ("public", "private"):
-                    if "is_public" in kwargs:
+                if FunctionVisibility.is_valid_value(decorator.id):
+                    if "function_visibility" in kwargs:
                         raise FunctionDeclarationException(
-                            "Visibility must be public or private, not both", node
+                            f"Visibility is already set to: {kwargs['function_visibility']}", node
                         )
-                    kwargs["is_public"] = bool(decorator.id == "public")
-                elif decorator.id == "payable":
-                    kwargs["is_payable"] = True
-                elif decorator.id == "view":
-                    kwargs["is_constant"] = True
+                    kwargs["function_visibility"] = FunctionVisibility(decorator.id)
+
+                elif StateMutability.is_valid_value(decorator.id):
+                    if "state_mutability" in kwargs:
+                        raise FunctionDeclarationException(
+                            f"Mutability is already set to: {kwargs['state_mutability']}", node
+                        )
```

Comment:
I think this check didn't exist before?

Reply:
I don't think so..  previously it was either `@constant` or nothing, so modifiying would be implied..  thus you couldn't decorate it both ways.